File: 卡码/代码随想录/1/03-哈希表/Go/02-两个数组交集/LeetCode_0349.go
//349. 两个数组的交集
package main

// import "github.com/emirpasic/gods/sets/hashset"

func intersection(nums1 []int, nums2 []int) []int {
	hash:=make(map[int]bool)
	// res:=hashset.New()
	res:=make(map[int]bool)
	for _,num:=range nums1 {
		hash[num]=true
	}
	for _,num:=range nums2 {
		if _,ok:=hash[num];ok {
			// res.Add(num)
			res[num]=true
		}
	}
	result:=make([]int,len(res))
	index:=0
	for key,_:=range res {
		result[index]=key
		index++
	}
	return result
}

File: 卡码/代码随想录/1/03-哈希表/Python/02-两个数组交集/LeetCode_0349.py
#349. 两个数组的交集
from typing import *

class Solution:
	def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
			hash=set(nums1)
			hash.intersection_update(nums2)
			#return list(hash)
			return list(set(nums1)&set(nums2))